BAM.rst

```diff
@@ -508,13 +508,19 @@ for example, to repeat barcode calling with different options.
   succinctly, it contains the integer pair :math:`B_F, B_R`.  Integer
   codes represent 0-based position in the FASTA file of barcodes.
 
-- The integer (``uint8``) ``bq`` tag contains the barcode call confidence.
+- The integer (``int``) ``bq`` tag contains the barcode call confidence.
   If the ``BarcodeQuality`` element of the header is set to ``Score``,
-  then the tag represents the sum of the calculated Smith-Waterman scores
-  that support the call in the ``bc`` tag across.  On the other hand,
-  if the value of the header-tag is ``Probability`` instead, then the
-  tag value is a the Phred-scaled posterior probability that the  barcode
-  call in ``bc`` is correct
+  then the tag represents the mean normalized sum of the calculated 
+  Smith-Waterman scores that support the call in the ``bc`` tag across all 
+  subreads. For each barcode, the sum of the Smith-Waterman score is normalized
+  by the length of the barcode times the match score; this provides an integer 
+  value between 0 - 100.
```

Comment:
"... and then multiplied by 100, and rounded,".  

I think this is OK.  I don't know how effective this will be at normalizing between different SW parameters (if we were to change them), but I think that is beyond the scope of what we are trying to do here.

Honestly I don't know how the int8 / int32 discrepancy happened and we may want to rectify that in the future.

+1, with the edit above